<?xml version="1.0"?>
<launch>
  <!-- Define the namespace of each quadrotor -->
  <arg name="namespace1" default="uav1"/>
  <arg name="namespace2" default="uav2"/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">  <!-- Open Gazebo using the empty_world.launch -->
    <arg name="paused" value="true"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

   <group ns="$(arg namespace1)">
     <include file="$(find my_hector_uavs)/launch/my_spawn_simplified_quadrotor.launch"> <!-- Call another launch file with my parameters -->
       <arg name="name" value="$(arg namespace1)" /> <!-- Robot name in Gazebo -->
       <arg name="tf_prefix" value="$(arg namespace1)" /> 
       <arg name="base_link_frame" value="$(arg namespace1)/base_link" /> <!-- base_link_frame is required since more than one robot will be used -->
       <arg name="model" value="$(find my_hector_uavs)/urdf/my_simplified_quadrotor.xacro" />
       <arg name="x" value="0.0" /> <!-- Initial posture (x,y,z,yaw) -->
       <arg name="y" value="-1.0" />
       <arg name="z" value="0.3" />
       <arg name="Y" value="0.0" />
     </include>
   </group>

  <group ns="$(arg namespace2)">
     <include file="$(find my_hector_uavs)/launch/my_spawn_simplified_quadrotor.launch">
       <arg name="name" value="$(arg namespace2)" />
       <arg name="tf_prefix" value="$(arg namespace2)" />
       <arg name="base_link_frame" value="$(arg namespace2)/base_link" />
       <arg name="model" value="$(find my_hector_uavs)/urdf/my_simplified_quadrotor.xacro" />
       <arg name="x" value="0.0" />
       <arg name="y" value="1.0" />
       <arg name="z" value="0.3" />
       <arg name="Y" value="0.0" />
     </include>
   </group>

</launch>
